src/msg/async/ProtocolV2.cc

```diff
@@ -743,7 +743,8 @@ CtPtr ProtocolV2::read(CONTINUATION_RXBPTR_TYPE<ProtocolV2> &next,
   next.node = std::move(buffer);
   ssize_t r = connection->read(len, buf,
     [&next, this](char *buffer, int r) {
-      if (unlikely(pre_auth.enabled) && r >= 0) {
+      assert(r > 0);
```

Comment:
what is this change for?
```c++
    case STATE_CONNECTION_ESTABLISHED: {
      if (pendingReadLen) {
        ssize_t r = read(*pendingReadLen, read_buffer, readCallback);
        if (r <= 0) { // read all bytes, or an error occured
          pendingReadLen.reset();
          char *buf_tmp = read_buffer;
          read_buffer = nullptr;
          readCallback(buf_tmp, r);
        }
        logger->tinc(l_msgr_running_recv_time,
            ceph::mono_clock::now() - recv_start_time);
        return;
      }
```

i found this in `AsyncConnection::process()`. if we have `assert(r > 0)`, will the program abort?